src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -422,6 +876,43 @@ public DateTime AddMilliseconds(double value)
             return Add(value, 1);
         }
 
+        /// <summary>
+        /// Returns a new <see cref="DateTime"/> that adds the specified number of microseconds to the value of this instance.
+        /// </summary>
+        /// <param name="value">
+        /// A number of whole and fractional microseconds.
+        /// The <paramref name="value"/> parameter can be negative or positive.
+        /// Note that this value is rounded to the nearest integer.
+        /// </param>
+        /// <returns>
+        /// An object whose value is the sum of the date and time represented
+        /// by this instance and the number of microseconds represented by <paramref name="value"/>.
+        /// </returns>
+        /// <remarks>
+        /// This method does not change the value of this <see cref="DateTime"/>. Instead, it returns a new <see cref="DateTime"/>
+        /// whose value is the result of this operation.
+        ///
+        /// The fractional part of value is the fractional part of a microsecond.
+        /// For example, 4.5 is equivalent to 4 microseconds and 50 ticks, where one microseconds = 10 ticks.
+        ///
+        /// The value parameter is rounded to the nearest integer.
+        /// </remarks>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// The resulting <see cref="DateTime"/> is less than <see cref="MinValue"/> or greater than <see cref="MaxValue"/>.
+        /// </exception>
+        public DateTime AddMicroseconds(double value)
+        {
+            int ticks = (int)(value * TicksPerMicrosecond);
```

Comment:
We need to ensure there is no overflow here. Also, we should cast it to `long` instead of int. We need carefully check the `value` to be inside specific range to ensure no overflow nor truncation when casting.

CC @tannergooding

Reply:
Please look at https://github.com/dotnet/runtime/pull/67666#pullrequestreview-935638394